Pocket/pocket-ios
# Core data - Refactor Space to use child contexts when needed

## Summary
- This PR does some Core Data related refactors:
    
## Implementation Details
- Update `Space`
        - Reorders methods by entity
        - Add `makeChildBackgroundContext()` to create a child background context to `backgroundContext`, to better isolate fetch/update operations
        - Add methods `updateSlate(_ :)` and `updateLineup(_ :)` to abstract Core Data logic away from `SlateService`. The above methods use a child context
        - Injects a context instead of hardcoding `backgroundContext` in fetch and update methods

> **Note**
       This is still WIP, there will still be some methods were we still use `backgroundContext`, this will change in future PRs.

 - Update `Pocket (iOS)` Scheme
        - Add the following launch args:
           - -com.apple.CoreData.SQLDebug 1 (disabled)
           - -com.Apple.CoreData.ConcurrencyDebug 1 (enabled)
       - Enables the following diagnostics
           - Thread Sanitizer
       - Disables the following diagnostics
           - Zombie Objects
- Update `Source`
    - Remove method `fetchSlate(_ :)` related implementations, calls and unit tests.
## Test Steps
* Checkout/build/run this branch
* Use the app as much as you can and check that it behaves as expected and does not crash. Test in particular:
    - Login/Logout with an account that has many items
    - Save/unsave, archive/unarchive, favorite/unfavorite
    - Scroll the list, navigate between tabs
    - Stop and relaunch the app several times
    - Go offline/online

## Optional tests
 - Do a raindance
 - Test in different weather conditions, outside, inside, night, day, climb a mountain and test, etc etc

## PR Checklist:
- [x] ~~Added Unit / UI tests~~ removed some tests due to refactor

## Screenshots
NA
